src/fides/api/migrations/hash_migration_mixin.py

```diff
@@ -0,0 +1,82 @@
+from abc import abstractmethod
+from typing import Any, List, Optional, Tuple
+
+from sqlalchemy import Boolean, Column, Index
+from sqlalchemy.ext.declarative import declared_attr
+
+from fides.api.migrations.hash_migration_tracker import HashMigrationTracker
+from fides.api.schemas.redis_cache import MultiValue
+
+
+class HashMigrationMixin:
+    """
+    A mixin that includes common logic for models that have hashed fields that need be be migrated from using bcrypt to SHA-256.
+    """
+
+    # set existing rows to false but set to true on create since we'll
+    # be using a new hashing function moving forward
+    is_hash_migrated = Column(Boolean, nullable=False, server_default="f", default=True)
+
+    @declared_attr
+    def __table_args__(cls: Any) -> Tuple:
+        """
+        Creates a dynamically named partial index on the table to track unmigrated rows.
+        The size of the index will shrink as more rows are migrated.
+        """
+        return (
+            Index(
+                f"idx_{cls.__tablename__}_unmigrated",  # type: ignore[attr-defined]
+                "is_hash_migrated",
+                postgresql_where=cls.is_hash_migrated.is_(False),  # type: ignore[attr-defined]
+            ),
+        )
+
+    @classmethod
+    @abstractmethod
+    def bcrypt_hash_value(
+        cls,
+        value: MultiValue,
+        encoding: str = "UTF-8",
+    ) -> Optional[str]:
+        """Hash value using bcrypt."""
+
+    @classmethod
+    @abstractmethod
+    def hash_value(
+        cls,
+        value: MultiValue,
+        encoding: str = "UTF-8",
+    ) -> Optional[str]:
+        """Hash value using SHA-256."""
```

Comment:
I set these as abstract because I expect any class that implements the `HashMigrationMixin` to have these two functions. Not sure if there is a better way to do this